torchscale/architecture/config.py

```diff
@@ -110,7 +113,7 @@ def __init__(self, **kwargs):
         self.no_output_layer = kwargs.pop("no_output_layer", False)
         self.layernorm_eps = kwargs.pop("layernorm_eps", 1e-5)
         # Text
-        self.vocab_size = kwargs.pop("vocab_size", -1)
+        self.vocab_size = kwargs.pop("vocab_size", 4000)
```

Comment:
@dtingey, @DrewGalbraith, @jeffeardley -- Any reason why we're manually setting a default vocab size here? There is another instance later in this file as well. If we do need to assign a default, could we change both to 32768 to keep consistent with other files? (For reference, a similar change was done in 0f918d4)

Reply:
Yes, we can change both to the multiple of two, it just means we need to retrain our tokenizer. I'm happy to do that this afternoon, so lets change it to 32768. 

As for if we need a default at all, it seems like a team philosophy question to me. I personally think it's better to abstract as much away as we can for the user, especially if we eventually share the tokenizer in our repo in the name of open sourcing, but some people might opine we should make the user aware of everything they're working with, thus needing to define all their own variables and provide their own tokenizer. So idk.

Reply:
I feel that it is weird to set only this one default (and `self.max_seq_len = kwargs.pop("max_seq_len", 512)`) in the Microsoft code while we have defaults used for the command-line arguments and we don't touch torchscale for any of the other variables. I've changed this back to -1 in [52538e4](https://github.com/DRAGNLabs/301r_retnet/pull/25/commits/52538e4cae1c44050718d93762bb5d293bfcd15a). If this is an issue, we can roll back this change and implement all the different parameters in torchscale.

In way of this comment, why was `self.max_seq_len = kwargs.pop("max_seq_len", 512)` also added?